### PR TITLE
parpack: update documentation and fix Exception

### PR DESCRIPTION
**Pull request by davydden**
- [x] run tests

### CODE REVIEW

File: include/deal.II/lac/parpack_solver.h [hunk 1]
```diff
@@ -487,6 +515,19 @@ void PArpackSolver<VectorType>::set_shift(const double s )
...
 template <typename VectorType>
 void PArpackSolver<VectorType>::
+set_initial_vector(const VectorType &vec)
+{
+  initial_vector_provided = true;
+  Assert (resid.size() == local_indices.size(),
+          ExcDimensionMismatch(resid.size(),local_indices.size()));
+  vec.extract_subvector_to (local_indices.begin(),
+                            local_indices.end(),
+                            &resid[0]);
```
Comment (bangerth) [inline include/deal.II/lac/parpack_solver.h]: not wrong, but out of curiosity, why not just copy the vector (or assign it)?

File: tests/arpack/step-36_parpack.cc [hunk 2]
```diff
@@ -286,6 +286,10 @@ void test ()
         mpi_communicator,
         additional_data);
     eigensolver.reinit(locally_owned_dofs);
+    {
+      eigenfunctions[0] = 1.;
+      eigensolver.set_initial_vector(eigenfunctions[0]);
+    }
```
Comment (bangerth) [inline tests/arpack/step-36_parpack.cc]: is there a good reason here and below to enclose the block in braces?

File: include/deal.II/lac/parpack_solver.h (hunk 1 above)
Comment (davydden) [inline include/deal.II/lac/parpack_solver.h]: later on i need to give to Parpack wrappers an address where resid starts, that's why resid is `vector<double>` and not `VectorType`.

Comment (bangerth) [inline include/deal.II/lac/parpack_solver.h]: Oh, I see. OK then.

**Comment by bangerth** [comment]
OK. Please squash the last into one of the first two patches and merge yourself.

**Pull request merged**